README.md

```diff
@@ -164,27 +189,68 @@ export default withOnyx({
 }, true)(App);
 ```
 
-### Dependent Onyx Keys and withOnyx()
+### Dependent Onyx Keys and useOnyx()
 Some components need to subscribe to multiple Onyx keys at once and sometimes, one key might rely on the data from another key. This is similar to a JOIN in SQL.
 
 Example: To get the policy of a report, the `policy` key depends on the `report` key.
 
 ```javascript
-export default withOnyx({
-    report: {
-        key: ({reportID) => `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
-    },
-    policy: {
-        key: ({report}) => `${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`,
-    },
-})(App);
+import React from 'react';
+import {useOnyx} from 'react-native-onyx';
+const ONYXKEYS = {
+    REPORT: 'report_1234',
+    POLICY: 'policy_'
+};
+
+const App = () => {
+    const [report] = useOnyx(ONYXKEYS.REPORT);
+    const [policy] = useOnyx(`${ONYXKEYS.POLICY}${report.policyID}`);
+    
+    return (
+        <View>
+            {/* Render with policy data */}
+        </View>
+    );
+};
+
+export default App;
 ```
 
 Background info:
 - The `key` value can be a function that returns the key that Onyx subscribes to
 - The first argument to the `key` function is the `props` from the component
 
-**Detailed explanation of how this is handled and rendered:**
+```javascript
+const App = ({reportID}) => {
+    const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`);
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`);
+
+    return (
+        <View>
+            {/* Render with policy data */}
+        </View>
+    );
+};
+
+export default App;
+```
+
+**Detailed explanation of how this is handled and rendered with `useOnyx()`:**
+
+1. The component mounts with a `reportID={1234}` prop.
+2. The `useOnyx` hook evaluates the mapping and subscribes to the key `reports_1234` using the `reportID` prop.
+3. The `useOnyx` hook fetches the data for the key `reports_1234` from Onyx and sets the state with the initial value (if provided).
+4. Since `policyID` is not defined yet, it defaults to `undefined`. The `useOnyx` hook subscribes to the key `policies_undefined`.
+5. The `useOnyx` hook reads the data and updates the state of the component:
+   - `report={{reportID: 1234, policyID: 1, ...rest of the object...}}`   - `policy={undefined}` (since there is no policy with ID `undefined`)
```

Comment:
```suggestion
   - `report={{reportID: 1234, policyID: 1, ...rest of the object...}}`
   - `policy={undefined}` (since there is no policy with ID `undefined`)
```